rio_tiler/io/xarray.py

```diff
@@ -335,17 +335,17 @@ def point(
         ds_lon, ds_lat = transform_coords(coord_crs, self.crs, [lon], [lat])
 
         if not (
-            (self.bounds[0] < ds_lon[0] < self.bounds[2])
-            and (self.bounds[1] < ds_lat[0] < self.bounds[3])
+            (self.bounds[0] < ds_lat[0] < self.bounds[2])
+            and (self.bounds[1] < ds_lon[0] < self.bounds[3])
         ):
             raise PointOutsideBounds("Point is outside dataset bounds")
 
-        x, y = rowcol(self.input.rio.transform(), ds_lon, ds_lat)
+        x, y = rowcol(self.input.rio.transform(), ds_lat, ds_lon)
```

Comment:
rowcol() takes `lon` (x values in coordinate reference system), `lat` (y values in coordinate reference system.) as input and return a tuple of rows and cols 

There is definitely an issue here but the fix should be 
```
y, x = rowcol(self.input.rio.transform(), ds_lon, ds_lat)
```